.github/workflows/main.yml

```diff
@@ -12,13 +12,41 @@ jobs:
 
     runs-on: ubuntu-latest
 
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Ruby
-      uses: ruby/setup-ruby@v1
-      with:
-        bundler-cache: true
-    - name: Install dependencies
-      run: bundle install
-    - name: Run tests
-      run: bundle exec rake
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set git configuration
+        run: |
+          git config --global init.defaultBranch main
+          git config --global user.name 'GitHub Actions'
+          git config --global user.email 'microsoft@example.com'
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: false
```

Comment:
I was hoping to get this to work with `bundler-cache: true` and without the below bundle install step but it wasn't working as advertised initially and I didn't look further into it.